index.js

```diff
@@ -52,6 +52,50 @@ function init() {
             process.exit(1)
         }
 
+        try {
+            // promisify APIs
+            var interfaces = {
+                  Nvim: nvim.constructor,
+                  Buffer: nvim.Buffer,
+                  Window: nvim.Window,
+                  Tabpage: nvim.Tabpage,
+                }
+
+            Object.keys( interfaces ).forEach( function( key ) {
+                if ( interfaces[key] ) {
+                    Object.keys( interfaces[key].prototype ).forEach( function( method ) {
+                      var oldMethod = interfaces[key].prototype[method],
+                          newMethod
+
+                      newMethod = function() {
+                            var args = Array.prototype.slice.call( arguments ),
+                                context = this
+
+                            if ( typeof args[ args.length - 1 ] === 'function' ) {
+                              return oldMethod.apply( context, args )
+                            }
+
+                            return new Promise( function( resolve, reject ) {
+                                args.push( function( err, result ) {
+                                    if ( err ) {
+                                        reject( err )
+                                    } else {
+                                        resolve( result )
+                                    }
+                                })
+
+                                oldMethod.apply( context, args )
+                            })
+                        }
+
+                        interfaces[key].prototype[method] = newMethod
+                    })
+                }
+            })
+        } catch ( err ) {
```

Comment:
I'm not exactly sure under what conditions the promisification will fail, other than if `Promise === undefined`. It might make more sense to simply not promisify if that's the case rather than guard it with a `try...catch`.